### PR TITLE
Custom functions registration ui

### PR DESCRIPTION
**Pull request by nico-bellante**

### CODE REVIEW

File: src/server/core/snippet.helper.ts [hunk 1]
```diff
@@ -119,3 +119,8 @@ export function getShareableYaml(rawSnippet: ISnippet, additionalFields: ISnippe
 export function isMakerScript(script: IContentLanguagePair) {
     return script.content.indexOf('Experimental.ExcelMaker') >= 0;
 }
+
+const isCustomFunctionRegex = /@customfunction/i;
+export function isCustomFunctionScript(script: IContentLanguagePair) {
```
Comment (nico-bellante) [inline src/server/core/snippet.helper.ts]: Oh, that friendly helper.  maybe you should use it elsewhere

File: src/server/interfaces.ts [hunk 2]
```diff
@@ -63,14 +63,11 @@ export interface IRunnerHandlebarsContext {
 }
 
 export interface ICustomFunctionsRegisterHandlebarsContext {
-    snippets: ICustomFunctionsRegistrationRelevantData[];
-    snippetsDataBase64: string;
-
+    metadata: any;
```
Comment (nico-bellante) [inline src/server/interfaces.ts]: name something like "visualMetadata" and give it a type

File: src/client/app/helpers/custom.functions.helper.ts [hunk 3]
```diff
@@ -1,30 +1,22 @@
...
-                catch {
-                    throw new Error(`Error parsing metadata for snippet "${snippet.name}"`);
-                }
-            });
+            .filter(snippet => snippet.script && isCustomFunctionRegex.test(snippet.script.content));
 
     let data: IRegisterCustomFunctionsPostData = {
         snippets: allSnippetsToRegisterWithPossibleDuplicate,
-        displayLanguage: getDisplayLanguage()
+        displayLanguage: getDisplayLanguage(),
```
Comment (Rickyp) [inline src/client/app/helpers/custom.functions.helper.ts]: For my understanding, why is the comma needed for the last property?

Comment (nico-bellante) [reply src/client/app/helpers/custom.functions.helper.ts]: If every line handles its own comma, then to delete a property, you can just delete the line. To add a property, you can just copy the line and modify the keys and values. 

I think (this)[https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8] summarizes it better.

Comment (nico-bellante) [reply src/client/app/helpers/custom.functions.helper.ts]: I also just like the uniform-ness of it ;P

File: src/client/app/containers/editor.mode.ts [hunk 4]
```diff
@@ -137,11 +142,20 @@ export class EditorMode {
         }
     }
 
+    static _customFunctionsRegex = /@customfunction/i;
```
[the remark this answers is not in the document]
Comment (nico-bellante) [reply src/client/app/containers/editor.mode.ts]: yeah i forgot one

File: src/client/app/helpers/custom.functions.helper.ts (hunk 3 above)
Comment (Rickyp) [reply src/client/app/helpers/custom.functions.helper.ts]: Sounds good to me!

**Pull request merged**